### PR TITLE
add atomic and timeout options

### PR DESCRIPTION
**Pull request by zumic96**
# Description

Adds atomic flag and timeout parameter for helm_release resource.
Both are defined in HELM variable set and can be overriden by specifying them in project variables in octopus.
`HELM_ATOMIC` - true or false value for atomic upgrade
`HELM_TIMEOUT` - number of seconds before timeout, default 300 (5min), added this because when we set atomic to true it sets wait to true and in case any upgrade takes too long it can easily be updated.

[1775](https://drivevariant.atlassian.net/browse/CLOUD-1775)

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
Prerequisites:
- update tgva [ref in actions-octopus](https://github.com/variant-inc/actions-octopus/blob/f/cloud-1755-atomic/action.yml#L10) 
- update actions-octopus [ref in project](https://github.com/variant-inc/demo-python-flask-variant-api/blob/demo/luka-2/.github/workflows/octo-push.yaml#L36)

1. test case
- crash container so helm upgrade fails, [commit](https://github.com/variant-inc/demo-python-flask-variant-api/blob/demo/luka-2/Dockerfile#L18)
- [GH actions](https://github.com/variant-inc/demo-python-flask-variant-api/actions/runs/2655763042)
- [Octopus](https://octopus.apps.ops-drivevariant.com/app#/Spaces-2/projects/demo-api-luka/deployments/releases/0.1.0-demo-luka-2-0001.299/deployments/Deployments-41578?activeTab=taskLog)

Results:
```
Error: release demo-api-luka failed, and has been rolled back due to atomic being set: timed out waiting for the condition
```
Helm history:
```
PS C:\Users\luka.baruskin> helm history demo-api-luka -n demo
REVISION        UPDATED                         STATUS          CHART                   APP VERSION     DESCRIPTION
1               Fri Jul  8 13:30:29 2022        superseded      variant-api-2.1.12                      Install complete
2               Mon Jul 11 08:07:11 2022        superseded      variant-api-2.1.12                      Upgrade complete
3               Mon Jul 11 08:17:40 2022        superseded      variant-api-2.1.12                      Upgrade complete
4               Mon Jul 11 12:58:32 2022        superseded      variant-api-2.1.12                      Upgrade complete
5               Tue Jul 12 09:39:35 2022        superseded      variant-api-2.1.12                      Upgrade complete
6               Tue Jul 12 09:49:24 2022        superseded      variant-api-2.1.12                      Upgrade complete
7               Tue Jul 12 09:54:49 2022        superseded      variant-api-2.1.12                      Upgrade complete
8               Tue Jul 12 10:33:26 2022        failed          variant-api-2.1.12                      Upgrade "demo-api-luka" failed: timed out waiting for the condition
9               Tue Jul 12 10:38:28 2022        deployed        variant-api-2.1.12                      Rollback to 7
```
 
[x] Sanity Testing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: live/apps/api/terraform.tfvars.json [hunk 1]
```diff
@@ -6,5 +6,7 @@
     "okta_org_name": "",
     "okta_base_url": "",
     "aws_resource_name_prefix": "#{AWS_IAM_PREFIX}",
-    "namespace": "#{Octopus.ProjectGroup.Name}"
+    "namespace": "#{Octopus.ProjectGroup.Name}",
+    "atomic": "#{HELM_ATOMIC}",
```
Comment (vibindaniel) [inline live/apps/api/terraform.tfvars.json:10]: I don't think variable substitution should be added here.

File: modules/apps/api/variables.tf [hunk 2]
```diff
@@ -58,4 +58,14 @@ variable "role_arn" {
 variable "tags" {
   type        = map(string)
   description = "Tags Output from Tags Module"
-}
\ No newline at end of file
+}
+
+variable "atomic" {
```
Comment (vibindaniel) [inline modules/apps/api/variables.tf:63]: Default should be true

File: modules/apps/api/variables.tf [hunk 3]
```diff
@@ -58,4 +58,14 @@ variable "role_arn" {
...
+  type        = bool
+  description = "If true, sets atomic flag on helm upgrade, if upgrade fails it reverts it"
+}
+
+variable "timeout" {
```
Comment (vibindaniel) [inline modules/apps/api/variables.tf:68]: Default to 10m

File: live/apps/api/terraform.tfvars.json (hunk 1 above)
Comment (zumic96) [reply live/apps/api/terraform.tfvars.json:10]: you meant to to set them as $Env:TF_VAR_atomic and $Env:TF_VAR_timeout on terragrunt apply step [here](https://github.com/variant-inc/cake-runner/blob/d46ad9c774781ba02ca28c59ca02fff0700f0a24/src/Variant.Cake.Runner/Octopus/OctopusHelper.cs#L104-L106) or something else?

Comment (zumic96) [reply live/apps/api/terraform.tfvars.json:10]: Added variables in [EKS Cluster variable set](https://octopus.apps.ops-drivevariant.com/app#/Spaces-2/library/variables/LibraryVariableSets-21?activeTab=variables) as we talked yesterday.
Will do it on all spaces before merging.

**Review by genghis-tuan — APPROVED** [review]
The logic honors HELM_TIMEOUT values passed in via Octopus. 
First deployed with the TVA package with a successful deployment. Then deployed an intentional failure. The atomic error is presented with an intentionally crashing deployment. The deployment is rolled-back to the previously successful deployment.

File: live/apps/api/terraform.tfvars.json (hunk 1 above)
Comment (zumic96) [reply live/apps/api/terraform.tfvars.json:10]: Variables added in EKS cluster variable set on all spaces.

**Pull request merged**